tests/test_group.rs

```diff
@@ -2,233 +2,242 @@ use openmls::prelude::*;
+
+        let alice_update_key_package_bundle = KeyPackageBundle::new(
+            &[ciphersuite.name()],
+            &alice_credential_bundle,
+            mandatory_extensions,
+        )
+        .unwrap();
+        let alice_update_key_package = alice_update_key_package_bundle.get_key_package();
+        assert!(alice_update_key_package.verify().is_ok());
+
+        // Alice creates a group
+        let group_id = [1, 2, 3, 4];
+        let mut group_alice = MlsGroup::new(
+            &group_id,
+            ciphersuite.name(),
+            alice_key_package_bundle,
+            GroupConfig::default(),
+        )
+        .unwrap();
+
+        // Alice proposes to add Bob with forced self-update
+        // Even though there are only Add Proposals, this should generated a path field on the Commit
+        let bob_add_proposal = group_alice.create_add_proposal(
             group_aad,
             &alice_credential_bundle,
-            epoch_proposals,
-            true, /* force self-update */
-        ) {
+            bob_key_package.clone(),
+        );
+        let epoch_proposals = vec![bob_add_proposal];
+        let (mls_plaintext_commit, _welcome_bundle_alice_bob_option, kpb_option) = match group_alice
+            .create_commit(
+                group_aad,
+                &alice_credential_bundle,
+                epoch_proposals,
+                true, /* force self-update */
+            ) {
             Ok(c) => c,
             Err(e) => panic!("Error creating commit: {:?}", e),
         };
-    let commit = match &mls_plaintext_commit.content {
-        MLSPlaintextContentType::Commit((commit, _)) => commit,
-        _ => panic!(),
-    };
-    assert!(commit.path.is_some());
-    assert!(commit.path.is_some() && kpb_option.is_some());
-
-    // Alice adds Bob without forced self-update
-    // Since there are only Add Proposals, this does not generate a path field on the Commit
-    // Creating a second proposal to add the same member should not fail, only committing that proposal should fail
-    let bob_add_proposal = group_alice_1234.create_add_proposal(
-        group_aad,
-        &alice_credential_bundle,
-        bob_key_package.clone(),
-    );
-    let epoch_proposals = vec![bob_add_proposal];
-    let (mls_plaintext_commit, welcome_bundle_alice_bob_option, kpb_option) = match group_alice_1234
-        .create_commit(
+        let commit = match &mls_plaintext_commit.content {
+            MLSPlaintextContentType::Commit((commit, _)) => commit,
+            _ => panic!(),
+        };
+        assert!(commit.path.is_some());
+        assert!(commit.path.is_some() && kpb_option.is_some());
+
+        // Alice adds Bob without forced self-update
+        // Since there are only Add Proposals, this does not generate a path field on the Commit
+        // Creating a second proposal to add the same member should not fail, only committing that proposal should fail
```

Comment:
Should we test here, that adding that proposal twice does indeed fail? Or will we do that in the course of #133?

Reply:
As mentioned, we need a lot more testing. I just wanted to get the baseline ready.